projects/storefrontlib/src/cms-components/cart/cart-details/cart-details.component.ts

```diff
@@ -37,20 +47,51 @@ export class CartDetailsComponent implements OnInit {
 
   constructor(
     protected cartService: CartService,
-    protected promotionService?: PromotionService
+    protected promotionService?: PromotionService,
+    protected selectiveCartService?: SelectiveCartService,
+    private authService?: AuthService,
+    private routingService?: RoutingService,
+    private featureConfig?: FeatureConfigService
   ) {}
 
   ngOnInit() {
     this.cart$ = this.cartService.getActive();
+
     this.entries$ = this.cartService
       .getEntries()
       .pipe(filter(entries => entries.length > 0));
-    this.cartLoaded$ = this.cartService.getLoaded();
+
+    if (this.isSaveForLaterEnabled()) {
+      this.cartLoaded$ = combineLatest([
+        this.cartService.getLoaded(),
+        this.selectiveCartService.getLoaded(),
+        this.authService.isUserLoggedIn(),
+      ]).pipe(
+        tap(([, , loggedIn]) => (this.loggedIn = loggedIn)),
+        map(([cartLoaded, sflLoaded, loggedIn]) =>
+          loggedIn ? cartLoaded && sflLoaded : cartLoaded
+        )
+      );
+    }
+    //TODO remove for #5958
+    else {
+      this.cartLoaded$ = this.cartService.getLoaded();
+    }
+    //TODO  remove for #5958 end
     this.orderPromotions$ = this.promotionService.getOrderPromotions(
```

Comment:
Could you add check here if `promotionService`. It might be not provided.

Reply:
it's introduced by issue #5670, but also updated too.